### PR TITLE
Phpstan

### PR DESCRIPTION
**Pull request by szepeviktor**
* [x] I have followed everything :)

@laurelfulford Static analysis is like adding 100 👀

I've fixed all errors on Level 4. You can go higher!

### CODE REVIEW

**Comment by szepeviktor** [comment]
I think `PHPStan\\WordPress\\Composer\\FixWpStubs::php73Polyfill` needs to be executed as core's `is_countable()` shim conflicts with a symfony polyfill.

```json
  "scripts": {
    "php73": "PHPStan\\WordPress\\Composer\\FixWpStubs::php73Polyfill",
```

and run `composer run-script php73`

File: phpstan.neon.dist [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#$ composer require --dev phpstan/phpstan-shim szepeviktor/phpstan-wordpress
+#$ vendor/bin/phpstan analyse
```
Comment (claudiulodro) [inline phpstan.neon.dist]: Can you please add these scripts/dependencies and the `php73` script you proposed to `composer.json`? That would make it much easier to get set up and run these tests in the future.

File: tests/bootstrap.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+define( 'WP_DEFAULT_THEME', '' );
+
+function is_amp_endpoint() { return true; }
```
Comment (claudiulodro) [inline tests/bootstrap.php]: Is there a way to do this without needing to redeclare a bunch of functions as stubs? It seems inelegant and difficult to maintain the tests if every function needs to be declared as a stub.

Comment (szepeviktor) [reply tests/bootstrap.php]: Static analysis works by figuring out types, so we need all referenced source code:

- Jetpack
- WooCommerce

I though it more sustainable to include 6 function stubs than 2 whole plugin stubs.
What do you think?

Comment (claudiulodro) [reply tests/bootstrap.php]: Yeah that makes sense. Thanks for clarifying. The stubs for references to other plugin's functions aren't super difficult to maintain, and we probably won't need to add too many of those. 

What about the `newspack_is_amp` and other `newspack_`-prefixed stubs? The static analyzer should be analyzing `functions.php` and analyze the real functions, shouldn't it? Can't we just load `functions.php`?

File: phpstan.neon.dist (hunk 1 above)
Comment (szepeviktor) [reply phpstan.neon.dist]: Done.

File: tests/bootstrap.php (hunk 2 above)
Comment (szepeviktor) [reply tests/bootstrap.php]: Currently we have these stubs:

- WP core stubs
- functions.php's stubs
- this file

Comment (claudiulodro) [reply tests/bootstrap.php]: Ah OK I see now. The WP stubs are loaded from https://github.com/GiacoCorsiglia/wordpress-stubs and this doesn't actually use the real functions. 👍

**Comment by laurelfulford** [comment]
Thank you, @szepeviktor and @claudiulodro!

**Comment by szepeviktor** [comment]
You're welcome.

**Pull request merged**